File: plusMinus.js
const plusMinus = (array) => {
  let positiveCount = 0;
  let negativeCount = 0;
  let zeroCount = 0;
  const n = array.length;

  for (let nums of array) {
    if (nums > 0) positiveCount++;
    if (nums < 0) negativeCount--;
    else {
      zeroCount++;
    }
  }

  console.log((positiveCount / n).toFixed(6));
  console.log((negativeCount / n).toFixed(6));
  console.log((zeroCount / n).toFixed(6));
};

let arr = [-4, 3, -9, 0, 4, 1];

console.log(plusMinus(arr));

File: plusMinus/PlusMinus.java
import java.util.ArrayList;
import java.util.List;

public class PlusMinus {
    public static void calculateRatios(List<Integer> arr) {
        int positiveCount = 0;
        int negativeCount = 0;
        int zeroCount = 0;
        int n = arr.size();

        for (int num : arr) {
            if (num > 0)
                positiveCount++;
            else if (num < 0)
                negativeCount++;
            else
                zeroCount++;
        }

        double positiveRatio = (double) positiveCount / n;
        double negativeRatio = (double) negativeCount / n;
        double zeroRatio = (double) zeroCount / n;

        System.out.printf("%.6f%n", positiveRatio);
        System.out.printf("%.6f%n", negativeRatio);
        System.out.printf("%.6f%n", zeroRatio);
    }

    public static void main(String[] args) {
        List<Integer> arr = new ArrayList<>();
        arr.add(-4);
        arr.add(3);
        arr.add(-9);
        arr.add(0);
        arr.add(4);
        arr.add(1);
        
        calculateRatios(arr);
    }
}
